### PR TITLE
2.function

### PR DESCRIPTION
**Pull request by magdalena-u**
chapter: Functions:
1/ minimum
2/ recursion
3/ bean counting

### CODE REVIEW

File: bean.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+const countBs = (string) => {
+    return countChar(string, "B")
+}
```
Comment (grzegorz-bielski) [inline bean.js]: You could omit `{}` braces and `return` in here

File: bean.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+let countChar = (string, char) => {
```
Comment (grzegorz-bielski) [inline bean.js]: If you are not reassigning anything it's better to use `const`, this way you are signaling to the reader, that the value won't be changed during program execution

File: bean.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    let count = 0;
+    for (let i = 0; i < string.length; i++) {
+        if (string[i] === char) {
+            count++
+        }
+    }
+    return count
+}
```
Comment (grzegorz-bielski) [inline bean.js]: no new line :P

File: min.js [hunk 4]
```diff
@@ -0,0 +1 @@
+const minimum = (a, b) => (a < b ? a : b);
```
Comment (grzegorz-bielski) [inline min.js]: I've seen this already (a different version), why don't add this to the previous PR.

File: recursion.js [hunk 5]
```diff
@@ -0,0 +1,7 @@
+const isEven = (number) => {
+    if (number === 1) {
+        return false
+    } else if (number === 0) {
+        return true
+    } else return isEven(number - 2)
+}
```
Comment (grzegorz-bielski) [inline recursion.js]: This doesn't need to be recursive, try to solve it using `modulo` (%) operator.

File: recursion.js [hunk 6]
```diff
@@ -0,0 +1 @@
+const isEven = (number) => (number % 2 === 0) ? true : false
```
Comment (grzegorz-bielski) [inline recursion.js]: You don't event need ternary here :P
```
const isEven = number => number % 2 === 0
```

**Review by grzegorz-bielski — APPROVED** [review]

File: bean.js [hunk 7]
```diff
@@ -0,0 +1,11 @@
+const countBs = string => countChar(string, 'B');
+
+const countChar = (string, char) => {
+    let count = 0;
+    for (let i = 0; i < string.length; i++) {
```
Comment (michal-lepik) [inline bean.js]: looks good, but try replacing the `for` loop with `forEach`

Comment (AdamusPiotr) [reply bean.js]: You can;t loop over string so try to use it like that

`[...string].forEach()`

or 

`string.split('').forEach()`

**Review by michal-lepik — APPROVED** [review]

**Pull request merged**